content/ideas/json-autotype-webapi.md

```diff
@@ -0,0 +1,13 @@
+---
+title: Expand json-autotype to directly query whole Web APIs, and create Servant endpoints
+---
+
+[json-autotype](http://github.com/mgajda/json-autotype) is currently the *most advanced* type and parser generator from JSON to Haskell.
+However most programmers are not using JSON just to read and write it, but to interact with Web APIs. One would dream about generating the whole WebAPI types, and either client calls, and server endpoints just from Swagger descriptions, or a bunch of queries to the WebAPIs.
+
+Implementing this project would put Haskell as possibly the best language to interact with WebAPIs in typed way. It would also make it superior to pure GraphQL solutions which are state-of-art now.
+
+**Mentor**: [Michal Gajda](mailto:migamake&#64;migamake.com)
+
+**Difficulty**: Advanced
+
```

Comment:
This project is a bit underspecified, and it at least needs information on what the advantages are over contributing to the _existing_ Haskell swagger toolchain, which I believe works fairly well.